src/components/dashboard/chores/Chores.js

```diff
@@ -0,0 +1,316 @@
+        this.getFrequencies();
+        M.AutoInit();
+    }
+
+    handleChoreClick(event, modal, chore) {
+        event.preventDefault();
+        let elem = document.querySelector(modal);
+        this.getUserById(chore.assignee_id);
+        this.currentChore = chore; 
+
+        if(!chore.notes) {
+            this.setState({ 
+                choreCategoryId: chore.category_id,
+                choreName: chore.chore_name,
+                choreDateComplete: chore.date_complete,
+                choreDifficulty: chore.difficulty,
+                choreFreqId: chore.frequency_id,
+                choreNotes: '',
+                choreStatus: chore.status,
+                detailsBool: !this.state.detailsBool
+            });
+        } else {
+            this.setState({ 
+                choreCategoryId: chore.category_id,
+                choreName: chore.chore_name,
+                choreDateComplete: chore.date_complete,
+                choreDifficulty: chore.difficulty,
+                choreFreqId: chore.frequency_id,
+                choreNotes: chore.notes,
+                choreStatus: chore.status,
+                detailsBool: !this.state.detailsBool
+            });
+        }
+
+        M.Modal.init(elem, {});
+        let instance = M.Modal.getInstance(elem);
+        instance.open();
+    }
+
+    handleChoreNameChange = (event) => {
+        this.setState({ choreName: event.target.value });
+    }
+
+    handleDifficultyChange = (event) => {
+        this.setState({ choreDifficulty: event.target.value });
+    }
+
+    handleCategoryChange = (event) => {
+        this.setState({ choreCategoryId: event.target.value });
+    }
+
+    handleFrequencyChange = (event) => {
+        this.setState({ choreFreqId: event.target.value });
+    }
+
+    handleStatusChange = (event) => {
+        if(event.target.value === "Completed") {
+            this.setState({ 
+                choreStatus: event.target.value,
+                choreDateComplete: new Date()
+            })
+        }
+        this.setState({ choreStatus: event.target.value });
+    }
+
+    handleNoteChange = (event) => {
+        this.setState({ choreNotes: event.target.value });
+    }
+
+    handleAssigneeChange = (event) => {
+        this.setState({ choreAssigneeUsername: event.target.value });
+    }
+
+    handleSaveEditChore(event, choreId, choreName, choreStatus, choreFreqId, choreCatId, choreAssigneeUsername, choreDifficulty, choreNotes) {
+        event.preventDefault();
+        let assigneeId = null;
+        
+        if(choreName && choreStatus && choreCatId) {
+            if(choreAssigneeUsername) {
+                fetch(`${baseUrl}/get-users`)
+                .then(res => {
+                    return res.json();
+                })
+                .then(users => {
+    
+                    users.map(user => {
+                        if(user.username === choreAssigneeUsername) {
+                            assigneeId = user.id;
```

Comment:
Look into how to break out of a map.  Maybe `return`?

Reply:
You can't break out of a map, the internet said that for that you should use a for loop instead so I switched it to that